Sources/TuistKit/Services/EditService.swift

```diff
@@ -56,27 +58,22 @@ final class EditService {
         let plugins = await loadPlugins(at: path)
 
         if !permanent {
-            try withTemporaryDirectory(removeTreeOnDeinit: true) { generationDirectory in
-                EditService.temporaryDirectory = generationDirectory
-
-                signalHandler.trap { _ in
-                    try? EditService.temporaryDirectory.map(FileHandler.shared.delete)
-                    exit(0)
-                }
-
-                guard let selectedXcode = try XcodeController.shared.selected() else {
-                    throw EditServiceError.xcodeNotSelected
-                }
-
-                let workspacePath = try projectEditor.edit(
-                    at: path,
-                    in: generationDirectory,
-                    onlyCurrentDirectory: onlyCurrentDirectory,
-                    plugins: plugins
-                )
-                logger.pretty("Opening Xcode to edit the project. Press \(.keystroke("CTRL + C")) once you are done editing")
-                try opener.open(path: workspacePath, application: selectedXcode.path, wait: true)
+            let pathHash = path.hashValue
```

Comment:
If I'm right, this is not deterministic across CLI runs. Therefore, I suggest taking a different approach. What about:
```
{basename}-{hash(path)}
```
The `basename` part is useful to identify the project, and the second part is to disambiguate in case there are two projects with the same name. For the hashed path you can use [MD5](https://github.com/tuist/tuist/blob/87cdf104245c120c89a8285e6af17bd61648b19a/Sources/TuistSupport/Extensions/String%2BMD5.swift#L167)

Reply:
Great idea!